src/coprocessor/dag/batch_executor/table_scan_executor.rs

```diff
@@ -310,482 +312,296 @@ mod tests {
+                }
+                assert_eq!(columns[id].decoded(), &values[col_idx]);
+            }
         }
+    }
 
-        // Case 9. PK is the last column in schema
-        {
-            let mut executor = BatchTableScanExecutor::new(
-                store.clone(),
-                Arc::new(EvalConfig::default()),
-                vec![
-                    columns_info[1].clone(),
-                    columns_info[2].clone(),
-                    columns_info[0].clone(),
-                ],
-                key_ranges_all.clone(),
-                false,
-            )
-            .unwrap();
+    fn test_basic_scan_wrapper(
+        store: &TableScanTestWrapper,
+        ranges: Vec<KeyRange>,
+        col_idxs: &[usize],
+    ) {
+        let columns_info = store.columns_info_by_idx(col_idxs);
+        let mut executor = BatchTableScanExecutor::new(
+            store.store(),
+            Arc::new(EvalConfig::default()),
+            columns_info.clone(),
+            ranges,
+            false,
+        )
+        .unwrap();
+
+        // First, let's fetch 1 row. There should be 4 rows remaining.
+        let expect_rows = 1;
+        let result = executor.next_batch(expect_rows);
+        assert!(!result.is_drained.as_ref().unwrap());
+        store.expect_table_values(col_idxs, 0, expect_rows, result.data);
+        // Then, fetch all remaining rows using a larger batch size.
+        let result = executor.next_batch(10);
+        let start_row = 1;
+        let expect_rows = 4;
+        // This time it should be drained.
+        assert!(result.is_drained.as_ref().unwrap());
+        store.expect_table_values(col_idxs, start_row, expect_rows, result.data);
+    }
 
-            let mut result = executor.next_batch(10);
-            assert!(result.is_drained.as_ref().unwrap());
-            assert_eq!(result.data.columns_len(), 3);
-            assert_eq!(result.data.rows_len(), 5);
-            assert!(result.data[0].is_raw());
-            result.data[0].decode(&Tz::utc(), &schema[1]).unwrap();
-            assert_eq!(
-                result.data[0].decoded().as_int_slice(),
-                &[Some(10), Some(-5), None, None, None]
-            );
-            assert!(result.data[1].is_raw());
-            result.data[1].decode(&Tz::utc(), &schema[2]).unwrap();
-            assert_eq!(
-                result.data[1].decoded().as_real_slice(),
-                &[Some(5.2), None, Some(4.5), Some(0.1), Some(4.5)]
-            );
-            assert!(result.data[2].is_decoded());
-            assert_eq!(
-                result.data[2].decoded().as_int_slice(),
-                &[Some(1), Some(3), Some(4), Some(5), Some(6)]
-            );
+    #[test]
+    fn test_basic_scan() {
+        let store = TableScanTestWrapper::default();
+        let test_ranges = vec![
+            store.point_ranges(),                 // point scan
+            vec![store.whole_table_range()],      // range scan
```

Comment:
`point_ranges`, `whole_table_range` and `mixed_ranges_for_whole_table` should be only useful in this test, which is the part of the test itself. Maybe not very meanful to be placed as `store` member functions.

Reply:
> A function is:
> 1. a block of code that performs a specific task. 
> 2. should be as small as possible.
> 3. Should do one thing only and do it well. 
> ...
>  --clean code

And we defined them in independent functions because we would like the reviewers can focus on what the range is while do not need to consider the implementation of how point_ranges or whole table range created.